script/grant_revoke_gem_authority.rb

```diff
@@ -1,82 +1,128 @@
+# Some GitHub user instances do not have an email address defined,
+# so start with the prior list of addresses (registered with Rubygems.org)
+KNOWN_COMMITTER_EMAIL_ADDRESSES = {
+  'awead' => "amsterdamos@gmail.com",
+  'barmintor' => "armintor@gmail.com",
+  'bess' => "bess@stanford.edu",
+  'cam156' => "cam156@psu.edu",
+  'cbeer' => "chris@cbeer.info",
+  'cjcolvar' => "cjcolvar@indiana.edu",
+  'DanCoughlin' => "dan.coughlin@gmail.com",
+  'dchandekstark' => "dchandekstark@gmail.com",
+  'jeremyf' => "jeremy.n.friesen@gmail.com",
+  'jkeck' => "jessie.keck@gmail.com",
+  'coblej' => "jim.coble@duke.edu",
+  'jpstroop' => "jpstroop@gmail.com",
+  'jcoyne' => "justin@curationexperts.com",
+  'mjgiarlo' => "leftwing@alumni.rutgers.edu",
+  'mark-dce' => "mark@curationexperts.com",
+  'mbklein' => "mbklein@gmail.com",
+  'mkorcy' => "mkorcy@gmail.com",
+  'ndushay' => "ndushay@stanford.edu",
+  'terrellt' => "trey.terrell@oregonstate.edu"
+}
+# Some GitHub repositories are named differently from their gems
+KNOWN_MISMATCHED_GEM_NAMES = {
+  'active_fedora' => 'active-fedora',
+  'fcrepo-admin' => 'fcrepo_admin',
+  'questioning_authority' => 'qa'
+}
+# GitHub repositories with matching gems that aren't from Hydra
+FALSE_POSITIVES = [
+  'hypatia'
+]
+# Gems that do not have their own GitHub repositories
+HANGERS_ON = [
   'hydra-core',
-  'hydra-derivatives',
-  'hydra-editor',
-  'hydra-head',
-  'hydra-ldap',
-  'hydra-migrate',
-  'hydra-mods',
-  'hydra-role-management',
-  'hydra-file_characterization',
-  'jettywrapper',
-  'curate',
-  'om',
-  'qa',
-  'rubydora',
-  'solrizer',
-  'solrizer-fedora',
-  'sufia',
-  'sufia-models',
+  'hydra-access-controls',
+  'sufia-models'
 ]
 
-HYDRA_COMMITTER_EMAILS = [
-  "amsterdamos@gmail.com",
-  "armintor@gmail.com",
-  "bess@stanford.edu",
-  "cam156@psu.edu",
-  "chris@cbeer.info",
-  "cjcolvar@indiana.edu",
-  "dan.coughlin@gmail.com",
-  "dchandekstark@gmail.com",
-  "edwin.shin@yourmediashelf.com",
-  "jeremy.n.friesen@gmail.com",
-  "jessie.keck@gmail.com",
-  "jim.coble@duke.edu",
-  "jpstroop@gmail.com",
-  "justin@curationexperts.com",
-  "leftwing@alumni.rutgers.edu",
-  "mark@curationexperts.com",
-  "matt@curationexperts.com",
-  "mbklein@gmail.com",
-  "mkorcy@gmail.com",
-  "mstrom81@gmail.com",
-  "ndushay@stanford.edu",
-]
+puts "(Hang in there! This script takes a couple minutes to run.)"
+
+github = Github.new(oauth_token: AUTHORIZATION_TOKEN, auto_pagination: true)
+
+# Get the ID of the GitHub-provided "Owners" team from the projecthydra org
+# We don't currently grant gem ownership to the folks in the other two orgs
+# (This just preserves the prior behavior.)
+owner_team_id = github.orgs.teams.list('projecthydra').select { |team| team.name == 'Owners' }.first.id
+owners = github.orgs.teams.list_members(owner_team_id)
+# Start with the prior (known to work) list of email addresses
+committer_map = KNOWN_COMMITTER_EMAIL_ADDRESSES
```

Comment:
I'm wondering about the constant usage on 63 and 67? Is there a reason for the mixture?